src/chain_spec.rs

```diff
@@ -390,13 +292,70 @@ fn genesis_processed_data(
     (identities, stakers, complete_txs)
 }
 
-fn balances(inits: &[InitialAuth], endoweds: &[AccountId]) -> Vec<(AccountId, u128)> {
-    endoweds
+fn dev_genesis_processed_data(
+    initial_authorities: &Vec<InitialAuth>,
+    key_bridge_locks: Vec<BridgeLockId>,
+    other_funded_accounts: Vec<AccountId>,
+) -> (
+    GenesisIdentityRecord<AccountId>,
+    Vec<(
+        IdentityId,
+        AccountId,
+        AccountId,
+        u128,
+        StakerStatus<AccountId>,
+    )>,
+    Vec<BridgeTx<AccountId, u128>>,
+) {
+    let mut identity = GenesisIdentityRecord::new(1u8, initial_authorities[0].0.clone());
+
+    let mut stakers = Vec::with_capacity(initial_authorities.len());
+    for (stash, controller, ..) in initial_authorities {
+        stakers.push((
+            IdentityId::from(1),
+            stash.clone(),
+            controller.clone(),
+            BOOTSTRAP_STASH / 2,
+            pallet_staking::StakerStatus::Validator,
+        ));
+        identity
+            .secondary_keys
+            .push(SecondaryKey::from_account_id_with_full_perms(stash.clone()));
+        identity
+            .secondary_keys
+            .push(SecondaryKey::from_account_id_with_full_perms(
+                controller.clone(),
+            ));
+    }
+
+    for account in other_funded_accounts {
+        identity
+            .secondary_keys
+            .push(SecondaryKey::from_account_id_with_full_perms(account));
```

Comment:
The number of pushes to `identity.secondary_keys` is known ahead of time, so let's use `.reserve()` before pushing.